File: Binary tree/Breath_First_Search/java/LevelOrderTraversal.java
class Node{
    int data;
    Node left ;
    Node right ;

    public Node(int data){
        this.data = data;
        left = null;
        right=null;
    }
}


class LevelOrderTraversal{
     public void levelOrderTraversal(Node root) {
        if (root == null) {
            return;
        }

        // Queue for traversing nodes level by level
        Queue<Node> queue = new LinkedList<>();
        queue.add(root);

        while (!queue.isEmpty()) {
            Node currentNode = queue.poll(); // Dequeue the current node
            System.out.print(currentNode.data + " "); // Print current node data

            // Enqueue left child
            if (currentNode.left != null) {
                queue.add(currentNode.left);
            }

            // Enqueue right child
            if (currentNode.right != null) {
                queue.add(currentNode.right);
            }
        }
     }


}

File: Binary tree/Breath_First_Search/js/LevelOrderTraversal.js
class Node {
  constructor(data) {
    this.data = data;
    this.left = null;
    this.right = null;
  }
}

class LevelOrderTraversal {
  levelOrder(root) {
    if (root === null) {
      return;
    }

    let queue = [];
    queue.push(root);

    while (queue.length > 0) {
      const number = queue.shift();
      console.log(number.data);

      if (number.left) {
        queue.push(number.left);
      }

      if (number.right) {
        queue.push(number.right);
      }
    }
  }
}

function main() {
  let root = new Node(1);
  root.left = new Node(2);
  root.right = new Node(3);
  root.left.left = new Node(4);
  root.left.right = new Node(5);
  root.right.left = new Node(6);
  root.right.right = new Node(7);

  let traversal = new LevelOrderTraversal();
  traversal.levelOrder(root);
}

main();
